mpas_analysis/ocean/climatology_map_waves.py

```diff
@@ -0,0 +1,526 @@
+           A list of seasons (keys in ``constants.monthDictionary``) over
+           which the climatology should be computed.
+
+        fileName : str
+            The name of the observation file
+
+        outFilePrefix : str
+            The prefix in front of output files and mapping files, typically
+            the name of the field being remapped
+
+        variableList : list
+            List of observational variables to remap
+
+        comparisonGridNames : list of {'latlon', 'antarctic', 'arctic'},
+            optional
+            The name(s) of the comparison grid to use for remapping.
+
+        subtaskName : str, optional
+            The name of the subtask
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        self.variableList = variableList
+
+        # call the constructor from the base class (AnalysisTask)
+        super().__init__(parentTask, seasons, fileName, outFilePrefix,
+                         comparisonGridNames=['latlon'],
+                         subtaskName=subtaskName)
+        # }}}
+
+    def get_observation_descriptor(self, fileName):  # {{{
+        '''
+        get a MeshDescriptor for the observation grid
+
+        Parameters
+        ----------
+        fileName : str
+            observation file name describing the source grid
+
+        Returns
+        -------
+        obsDescriptor : ``MeshDescriptor``
+            The descriptor for the observation grid
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        # create a descriptor of the observation grid using the lat/lon
+        # coordinates
+        obsDescriptor = LatLonGridDescriptor.read(fileName=fileName,
+                                                  latVarName='latitude',
+                                                  lonVarName='longitude')
+        return obsDescriptor  # }}}
+
+    def build_observational_dataset(self, fileName):  # {{{
+        '''
+        read in the data sets for observations, and possibly rename some
+        variables and dimensions
+
+        Parameters
+        ----------
+        fileName : str
+            observation file name
+
+        Returns
+        -------
+        dsObs : ``xarray.Dataset``
+            The observational dataset
+        '''
+        # Authors
+        # -------
+        # Steven Brus
+
+        sectionName = self.taskName
+        climStartYear = self.config.getint(sectionName, 'era5ObsStartYear')
+        climEndYear = self.config.getint(sectionName, 'era5ObsEndYear')
+        timeStart = datetime.datetime(year=climStartYear, month=1, day=1)
+        timeEnd = datetime.datetime(year=climEndYear, month=12, day=31)
+
+        dsObs = xr.open_dataset(fileName)
+        dsObs = dsObs.rename({'time': 'Time'})
+        dsObs = dsObs.sel(Time=slice(timeStart, timeEnd))
+        dsObs.coords['month'] = dsObs['Time.month']
+        dsObs.coords['year'] = dsObs['Time.year']
+
+        dsObs = dsObs[self.variableList]
+
```

Comment:
Similar to BGC:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/846917e695e59a672bdb256ff2156e555b8c06e8/mpas_analysis/ocean/climatology_map_bgc.py#L373-L374
I think you want something like:
```suggestion

        degrees = 'degree' in dsObs.lon.units
        dsObs = add_periodic_lon(ds=dsObs, lonDim='lon', degrees=degrees)
```

Reply:
For this to work, we will also need to fix a bug.  This line:
https://github.com/MPAS-Dev/MPAS-Analysis/blob/develop/mpas_analysis/shared/interpolation/utility.py#L22
should instead be:
``` python
    lonRange = ds[lonDim][-1].values - ds[lonDim][0].values
```

Reply:
This is great; thanks for the help with this.

Reply:
@xylar, I'm seeing this warning:

```
/gpfs/fs1/home/sbrus/MPAS-Analysis/mpas_analysis/shared/interpolation/utility.py:37: UserWarning: rename 'newLon' to 'longitude' does not create an index anymore. Try using swap_dims instead or use set_index after rename to create an indexed coordinate.
  ds = ds.rename({'newLon': lonDim})
```

Is this something you want to try to avoid?

Reply:
I saw that, too.  This looks to me like a warning from xarray and I think we can ignore it.